src/workflow/unlock.c

```diff
@@ -132,6 +145,12 @@ void workflow_unlock_enter_done(const char* password)
 
 void workflow_unlock(void)
 {
-    // "Enter password"
-    ui_screen_stack_switch(entry_screen_create("Enter password", _enter));
+    if (!memory_is_initialized() || keystore_is_unlocked()) {
```

Comment:
Is this a sanity check or are there valid code paths that lead to this return? If sanity-check, should it debug print or abort?

Reply:
It is a valid code path. OP_UNLOCK calls this, and it does nothing if we are already unlocked (can happen when if you restart the desktop app or switch wallets while the device is is unlocked).

Reply:
OK